Assignment #5

In [1]:
import pandas as pd
import numpy as np

Question 1- What is the northernmost airport in the United States?
First thing we do is to read the data in panads. 


In [24]:
airports = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv')
airports.head()

faa                           name        lat        lon   alt   tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044 -5.0   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264 -6.0   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801 -6.0   A   
3  06N                Randall Airport  41.431912 -74.391561   523 -5.0   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11 -5.0   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

To figure out the northern most airport, we have to look at the data with the most latitiude. Latitudes of +90 and -90 degrees correspond to the north and south geographic poles on the earth, respectively

In [40]:
airports.sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt   tz dst              tzone  
417  149 -5.0   A                 \N  
230   44 -9.0   A  America/Anchorage  
110   41 -9.0   A  America/Anchorage  
708   35 -9.0   A  America/Anchorage  
152   96 -9.0   A  America/Anchorage

Answer - Based on the latitude, the northern most airport is Wiley Post Will Rogers Memorial. The tzone of Dillant Hopkins Airport is unknown, so we cannot determine if it is in America or not based on the data that is given. 

Question 2- What is the easternmost airport in the United States?
The longitude is distance east or west of the prime meridian (an imaginary line running from north to south through Greenwich, England).

In [35]:
airports.sort_values('lon', ascending = False).head()

faa                           name        lat         lon   alt   tz  \
1290  SYA                   Eareckson As  52.712275  174.113620    98 -9.0   
942   MYF               Montgomery Field  32.475900  117.759000    17  8.0   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478  8.0   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149 -5.0   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45 -5.0   

     dst              tzone  
1290   A  America/Anchorage  
942    A     Asia/Chongqing  
396    A     Asia/Chongqing  
417    A                 \N  
444    A   America/New_York

The easternmost airport is Eastport Municipal Airport in New York. Eareckson As airport is in America but it is located on the western part of America. 

Question 3 - On February 12th, 2013, which New York area airport had the windiest weather?

In [41]:
weather = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')
weather.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     0  37.04  21.92  53.97     230.0    10.35702   
1    EWR  2013      1    1     1  37.04  21.92  53.97     230.0    13.80936   
2    EWR  2013      1    1     2  37.94  21.92  52.09     230.0    12.65858   
3    EWR  2013      1    1     3  37.94  23.00  54.51     230.0    13.80936   
4    EWR  2013      1    1     4  37.94  24.08  57.04     240.0    14.96014   

   wind_gust  precip  pressure  visib             time_hour  
0  11.918651     0.0    1013.9   10.0  2013-01-01T00:00:00Z  
1  15.891535     0.0    1013.0   10.0  2013-01-01T01:00:00Z  
2  14.567241     0.0    1012.6   10.0  2013-01-01T02:00:00Z  
3  15.891535     0.0    1012.7   10.0  2013-01-01T03:00:00Z  
4  17.215830     0.0    1012.8   10.0  2013-01-01T04:00:00Z

In [43]:
speediest = weather[(weather.month == 2) & (weather.day == 12) & (weather.year == 2013)]
speediest.sort_values('wind_speed', ascending = False).head()


origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

The airport with the windiest weather is EWR on Feburary 12th, 2013. The wind speed was 1048.36058. This was done by first gathering all the wind speed data on Feburary 12th then sorting the values based on wind speed. 